### PR TITLE
Fixed audio-video synchronisation problem in read_video() when using `pts` as unit

### PR DESCRIPTION
**Pull request by prabhat00155**

### CODE REVIEW

**Review by bjuncek — APPROVED** [review]
Looks good to me.
I've just put minor style edits.

Also, we should consider adding tests for this case to be able to catch the error in the future

File: torchvision/io/video.py [hunk 1]
```diff
@@ -278,6 +278,14 @@ def read_video(
 
     try:
         with av.open(filename, metadata_errors="ignore") as container:
+            time_base = None
+            if container.streams.video:
+                time_base = container.streams.video[0].time_base
+            elif container.streams.audio:
+                time_base = container.streams.audio[0].time_base
+            if pts_unit == 'pts':
+                start_pts, end_pts, pts_unit = _video_opt._convert_to_sec(
```
[the remark this answers is not in the document]
Comment (prabhat00155) [reply torchvision/io/video.py:287]: I see this naming convention being followed everywhere(with `pts_unit` being used to tell the difference). Perhaps, we should name it `start_offset`, `end_offset` and `offset_unit`. But then, it's a user-facing API.

File: torchvision/io/_video_opt.py [hunk 2]
```diff
@@ -485,6 +493,26 @@ def _read_video(filename, start_pts=0, end_pts=None, pts_unit="pts"):
...
+        )
+        time_base = video_timebase
+
+    if has_audio:
+        audio_timebase = Fraction(
+            info.audio_timebase.numerator, info.audio_timebase.denominator
+        )
+        time_base = time_base if time_base else audio_timebase
+
+    start_pts, end_pts, pts_unit = _convert_to_sec(
```
[the remark this answers is not in the document]
Comment (prabhat00155) [reply torchvision/io/_video_opt.py:514]: Do you mean add a comment or a docstring to the function _read_video()?

[the remark this answers is not in the document]
Comment (bjuncek) [reply torchvision/io/_video_opt.py:514]: adding a comment above line 514 should suffice I think

File: test/test_video_reader.py [hunk 3]
```diff
@@ -1241,6 +1241,45 @@ def test_read_video_from_memory_scripted(self):
...
+                    if index == cutoff:
+                        start_pts_val = frame.pts
+                    if index >= cutoff:
+                        arr.append(frame.to_rgb().to_ndarray())
+                visual, _, info = io.read_video(full_path, start_pts=start_pts_val, pts_unit='pts')
+                self.assertAlmostEqual(
+                    config.video_fps, info['video_fps'], delta=0.0001
+                )
+                arr = torch.Tensor(arr)
+                if arr.shape == visual.shape:
```
Comment (bjuncek) [inline test/test_video_reader.py:1266]: Why is this necessary? 
Wouldn't we expect them to be the same shape?

File: test/test_video_reader.py [hunk 4]
```diff
@@ -1241,6 +1241,45 @@ def test_read_video_from_memory_scripted(self):
...
+            container = av.open(full_path)
+            if container.streams.audio:
+                audio = container.streams.audio[0]
+                arr = []
+                for index, frame in enumerate(container.decode(audio)):
+                    if index >= cutoff:
+                        arr.append(frame.to_ndarray())
+                _, audio, _ = io.read_video(full_path, start_pts=start_pts_val, pts_unit='pts')
+                arr = torch.as_tensor(np.concatenate(arr, axis=1))
+                if arr.shape == audio.shape:
```
Comment (bjuncek) [inline test/test_video_reader.py:1279]: As above

**Review by bjuncek — COMMENTED** [review]
Looks good to me. Can you just clarify the above comment?

File: test/test_video_reader.py (hunk 3 above)
Comment (prabhat00155) [reply test/test_video_reader.py:1266]: There are some off by 1, which may be related to the differences in how the closest frame is found in pyav and torchvision(as we had discussed offline).
